lib/plan/setting_panel/setting_panel.dart

```diff
@@ -0,0 +1,284 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:trufi_core/widgets/custom_expanded_tile.dart';
+import 'package:trufi_core/widgets/custom_switch_tile.dart';
+
+import '../../entities/plan_entity/plan_entity.dart';
+
+import 'setting_panel_cubit.dart';
+
+class SettingPanel extends StatelessWidget {
+  static const String route = "/setting-panel";
+  const SettingPanel({Key key}) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    final theme = Theme.of(context);
+    // TODO to add all translates this widget
+    // final localization = TrufiLocalization.of(context);
+
+    final settingPanelCubit = context.read<SettingPanelCubit>();
+    return Scaffold(
+      appBar: AppBar(
+        title: const Text("Settings"),
+        leading: BackButton(
+          onPressed: () {
+            Navigator.of(context).pop(settingPanelCubit.state);
+          },
+        ),
+      ),
+      body: SafeArea(
+        child: BlocBuilder<SettingPanelCubit, SettingPanelState>(
+          builder: (blocContext, state) {
+            return ListView(
+              children: <Widget>[
+                const SizedBox(
+                  height: 10,
+                ),
+                CustomExpansionTile(
+                  title: "Walking speed",
+                  options: const [
+                    'Slow',
+                    'Calm',
+                    'Average',
+                    'Prompt',
+                    'Fast',
+                  ],
+                  textSelected: _getTextWalkingSelect(context,state.typeWalkingSpeed),
+                  onChanged: (value) {
+                    switch (value) {
+                      case 'Slow':
+                        settingPanelCubit.setWalkingSpeed(WalkingSpeed.slow);
+                        break;
+                      case 'Calm':
+                        settingPanelCubit.setWalkingSpeed(WalkingSpeed.calm);
+                        break;
+                      case 'Average':
+                        settingPanelCubit.setWalkingSpeed(WalkingSpeed.average);
+                        break;
+                      case 'Prompt':
+                        settingPanelCubit.setWalkingSpeed(WalkingSpeed.prompt);
+                        break;
+                      case 'Fast':
+                        settingPanelCubit.setWalkingSpeed(WalkingSpeed.fast);
+                        break;
+                      default:
+                    }
+                  },
+                ),
+                const Divider(thickness: 2),
+                CustomSwitchTile(
+                  title: 'Avoid walking',
+                  value: state.avoidWalking,
+                  onChanged: settingPanelCubit.setAvoidWalking,
+                ),
```

Comment:
Not sure but this could be a list of objects and then we use the ListView.seperated to create the different once.

Reply:
There will also be changes to the user interface, it would be good to wait.